src/table/utils/__tests__/selections-util.spec.ts

```diff
@@ -6,50 +7,59 @@ import {
   getCellSelectionState,
   SelectionStates,
 } from '../selections-utils';
+import {
+  TableCell,
+  SelectionState,
+  ExtendedSelectionAPI,
+  AnnounceFn,
+  SelectedRows,
+  ContextValue,
+  Action,
+} from '../../../types';
 
 describe('selections-utils', () => {
   describe('addSelectionListeners', () => {
     const listenerNames = ['deactivated', 'canceled', 'confirmed', 'cleared'];
-    let api;
-    let selectionDispatch;
-    let setShouldRefocus;
-    let keyboard;
-    let containsActiveElement;
-    let tableWrapperRef;
+    let api: ExtendedSelectionAPI;
+    let selectionDispatch: jest.Mock<any, any>;
+    let setShouldRefocus: jest.Mock<any, any>;
+    let keyboard: stardust.Keyboard;
+    let containsActiveElement: boolean;
+    let tableWrapperRef: React.MutableRefObject<HTMLElement>;
 
     beforeEach(() => {
       selectionDispatch = jest.fn();
       setShouldRefocus = jest.fn();
       api = {
         on: jest.fn(),
         removeListener: jest.fn(),
-      };
+      } as unknown as ExtendedSelectionAPI;
```

Comment:
Doing a bunch of these. Not sure if it is good practice. But for the tests it feels overkill do define everything, especially for events. Could also allow a simpler type in the arguments for easier mocking, but that just feels like it would be less strictly types -> not as effective catching wrong types

Reply:
Yeah I agree. Especially also since you're mocking everything.

Reply:
So this is fine?

Reply:
Depends on your level of ambition. Since it's a test file I would say it's fine.

Reply:
Yeah I would never do this in the actual code, kind of defeats the point